pkg/controller/clusterconnection/clusterconnection_controller.go

```diff
@@ -231,111 +224,89 @@ func (r *ReconcileConnection) Reconcile(ctx context.Context, request reconcile.R
 	// Changes for updating ManagementClusterConnection status conditions.
 	if request.Name == ResourceName && request.Namespace == "" {
 		ts := &operatorv1.TigeraStatus{}
-		err := r.Client.Get(ctx, types.NamespacedName{Name: ResourceName}, ts)
+		err := r.cli.Get(ctx, types.NamespacedName{Name: ResourceName}, ts)
 		if err != nil {
 			return reconcile.Result{}, err
 		}
 		managementClusterConnection.Status.Conditions = status.UpdateStatusCondition(managementClusterConnection.Status.Conditions, ts.Status.Conditions)
-		if err := r.Client.Status().Update(ctx, managementClusterConnection); err != nil {
+		if err := r.cli.Status().Update(ctx, managementClusterConnection); err != nil {
 			log.WithValues("reason", err).Info("Failed to create ManagementClusterConnection status conditions.")
 			return reconcile.Result{}, err
 		}
 	}
 
-	if managementClusterConnection != nil && managementCluster != nil {
-		err = fmt.Errorf("having both a ManagementCluster and a ManagementClusterConnection is not supported")
-		r.status.SetDegraded(operatorv1.ResourceValidationError, "", err, reqLogger)
-		return reconcile.Result{}, err
-	}
+	// Verify the cluster doesn't also have the ManagementCluster CRD installed.
+	if variant == operatorv1.TigeraSecureEnterprise {
+		managementCluster, err := utils.GetManagementCluster(ctx, r.cli)
+		if err != nil {
+			r.status.SetDegraded(operatorv1.ResourceReadError, "Error reading ManagementCluster", err, reqLogger)
+			return reconcile.Result{}, err
+		}
 
-	preDefaultPatchFrom := client.MergeFrom(managementClusterConnection.DeepCopy())
-	fillDefaults(managementClusterConnection)
+		if managementCluster != nil {
+			err = fmt.Errorf("having both a ManagementCluster and a ManagementClusterConnection is not supported")
+			r.status.SetDegraded(operatorv1.ResourceValidationError, "", err, reqLogger)
+			return reconcile.Result{}, err
+		}
+	}
 
-	// Write the discovered configuration back to the API. This is essentially a poor-man's defaulting, and
-	// ensures that we don't surprise anyone by changing defaults in a future version of the operator.
-	if err := r.Client.Patch(ctx, managementClusterConnection, preDefaultPatchFrom); err != nil {
+	if err := utils.ApplyDefaults(ctx, r.cli, managementClusterConnection); err != nil {
 		r.status.SetDegraded(operatorv1.ResourceUpdateError, err.Error(), err, reqLogger)
 		return reconcile.Result{}, err
 	}
```

Comment:
We should have a validation step here that verifies the MCC CA type is set to Tigera and not Public

Reply:
Does this comment belong here, for applying the defaults, or down below where I took out the check.

Reply:
it may not belong anywhere, I didn't see this logic but if it already exists then we're all good! I just threw it somewhere close to where I thought it should be.

Reply:
I updated the code to use the create the trusted bundle with the system cert since that's what it was doing before.

Reply:
also added the check to not allow public if this is calico.